File: backend-java/src/main/java/com/aiinterview/model/Interview.java
package com.aiinterview.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@Entity
@Table(name = "interviews")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EntityListeners(AuditingEntityListener.class)
public class Interview {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "candidate_id", nullable = false)
    private User candidate;
    
    @Enumerated(EnumType.STRING)
    @Column(name = "interview_type", nullable = false)
    private InterviewType interviewType;
    
    @Column(nullable = false)
    private Integer score;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private InterviewStatus status;
    
    @Enumerated(EnumType.STRING)
    @Column(name = "pipeline_stage")
    private PipelineStage stage;
    
    @Column(name = "feedback_summary", columnDefinition = "TEXT")
    private String feedbackSummary;
    
    @Column(name = "technical_score")
    private Integer technicalScore;
    
    @Column(name = "communication_score")
    private Integer communicationScore;
    
    @Column(name = "confidence_score")
    private Integer confidenceScore;
    
    @Column(name = "duration_minutes")
    private Integer durationMinutes;
    
    @Column(name = "questions_answered")
    private Integer questionsAnswered;
    
    @Column(name = "detailed_feedback", columnDefinition = "TEXT")
    private String detailedFeedback;  // JSON string
    
    @CreatedDate
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;
    
    @LastModifiedDate
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    
    public enum InterviewType {
        HR, TECHNICAL, COMBINED
    }
    
    public enum InterviewStatus {
        PENDING_REVIEW, REVIEWED, COMPLETED
    }
    
    public enum PipelineStage {
        APPLIED, SCREENING, ASSESSMENT, INTERVIEW, OFFER, HIRED
    }
}

File: flask/Emotion_detection_with_CNN-main/models/interview.py
from datetime import datetime
from extensions import db

class Interview(db.Model):
    """Interview/Result model for storing candidate interview data"""
    __tablename__ = 'interviews'
    
    id = db.Column(db.Integer, primary_key=True)
    candidate_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=False)
    interview_type = db.Column(db.String(20), nullable=False)  # 'hr' or 'technical'
    score = db.Column(db.Integer, nullable=False)  # 0-100
    feedback_summary = db.Column(db.Text, nullable=True)
    status = db.Column(db.String(20), nullable=False, default='pending_review')  # 'reviewed' or 'pending_review'
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    
    # Relationship to User
    candidate = db.relationship('User', backref='interviews')
    
    def to_dict(self):
        """Convert interview to dictionary"""
        return {
            'id': self.id,
            'candidate_id': self.candidate_id,
            'name': self.candidate.name if self.candidate else 'Unknown',
            'interview_type': self.interview_type,
            'score': self.score,
            'feedback_summary': self.feedback_summary,
            'status': self.status,
            'created_at': self.created_at.isoformat()
        }
    
    def __repr__(self):
        return f'<Interview {self.id}: {self.interview_type} for {self.candidate_id}>'